node/pkg/watchers/ibc/wasm_attrs.go

```diff
@@ -0,0 +1,102 @@
+package ibc
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strconv"
+
+	"github.com/tidwall/gjson"
+
+	"go.uber.org/zap"
+)
+
+// WasmAttributes is an object to facilitate parsing of wasm event attributes. It provides a method to parse the attribute array in a wasm event,
+// plus methods to return attributes as the appropriate type, including doing range checking.
+type WasmAttributes struct {
+	m map[string]string
+}
+
+// GetAsBytes returns the attribute value as a string.
+func (wa *WasmAttributes) GetAsString(key string) (string, error) {
+	value, exists := wa.m[key]
+	if !exists {
+		return "", fmt.Errorf("attribute %s does not exist", key)
+	}
+
+	return value, nil
+}
+
+// GetAsBytes returns the attribute value as an unsigned int. It also performs range checking.
+func (wa *WasmAttributes) GetAsUint64(key string, bitSize int) (uint64, error) {
```

Comment:
Wouldn't the name seem to imply that the bitSize can only be 64?

Reply:
Renamed the two functions to drop the "64".